packages/grid/src/GridRenderer.ts

```diff
@@ -1178,8 +1259,35 @@ export class GridRenderer {
         fontWidth,
         truncationChar
       );
+
+      // To check for links, we should check to the first space after the truncatedText lengh
+      let lengthOfContent = text.indexOf(' ', truncatedText.length);
+      // If it doesn't exist, set lengthOfContent to the minimum between length of the original text and 5000
+      if (lengthOfContent === -1) {
+        lengthOfContent = Math.min(5000, text.length);
+      } else if (lengthOfContent > 5000) {
+        // If the index is greater than 5000, limit it to 5000
+        lengthOfContent = 5000;
+      }
+      const contentToCheckForLinks = text.substring(0, lengthOfContent);
+
+      const links = model.findLinksInText(contentToCheckForLinks);
```

Comment:
We should probably consider memoizing based on `modelColumn`/`modelRow` instead of the content strings. Seems expensive to do implicit comparisons on up to 10k strings on each lookup. Then `indexOf` and `substring` could also be cached.

Reply:
How about moving the `indexOf` and `substring` into the cached function instead? So I pass in the original text in the cell and the truncated text. I'm not sure how to do it with just `modelColumn` and `modelRow` because `GridModel` doesn't have a method to get the truncated text of a cell.